### PR TITLE
Exception handlers & Modifiable timeout

### PR DESCRIPTION
**Pull request by TobiasG-DE**
This PR implements two enhancements for the current Network library:
  - Developers are now able to assign exception handlers as Consumer<Throwable> to RakNetClients and RakNetServers. They are
    called whenever an exception occurs in the pipeline.

  - Developers are able to set the timeout of a RakNetSession, which is SESSION_TIMEOUT_MS by default.

### CODE REVIEW

File: raknet/src/main/java/com/nukkitx/network/raknet/RakNetClient.java [hunk 1]
```diff
@@ -167,6 +182,9 @@ public void handlerAdded(ChannelHandlerContext ctx) {
         @Override
         public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
             log.error("An exception occurred in RakNet", cause);
```
Comment (Alemiz112) [inline raknet/src/main/java/com/nukkitx/network/raknet/RakNetClient.java:184]: What about not logging exception if `exceptionHandlers` is not empty. Handlers could care about logging too.

File: raknet/src/main/java/com/nukkitx/network/raknet/RakNetClient.java [hunk 2]
```diff
@@ -79,6 +85,15 @@ public RakNetClientSession create(InetSocketAddress address) {
         return pongFuture;
     }
 
+    public void addExceptionHandler(Consumer<Throwable> handler) {
+        Objects.requireNonNull(handler, "clientExceptionHandler");
+        this.exceptionHandlers.add(handler);
+    }
+
+    public void clearExceptionHandlers() {
```
[the remark this answers is not in the document]
Comment (TobiasG-DE) [reply raknet/src/main/java/com/nukkitx/network/raknet/RakNetClient.java:94]: I would, especially makes sense with the latest commit. Maybe you wanna remove the default exception handler before adding own ones. Another way I'm considering is to let the user give the handlers IDs, which they can use to remove specific handlers. I would then map <String, Consumer<Throwable>>

File: raknet/src/main/java/com/nukkitx/network/raknet/RakNetClient.java (hunk 1 above)
Comment (TobiasG-DE) [reply raknet/src/main/java/com/nukkitx/network/raknet/RakNetClient.java]: In the latest commit I changed that so logging is the default exception handler, but you can clear it to remove it and add own ones

File: raknet/src/main/java/com/nukkitx/network/raknet/RakNetClient.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Alemiz112) [reply raknet/src/main/java/com/nukkitx/network/raknet/RakNetClient.java:94]: Adding way to remove only one exception handler from list would be more useful.

File: raknet/src/main/java/com/nukkitx/network/raknet/RakNetClient.java [hunk 3]
```diff
@@ -20,6 +24,8 @@
     private static final InternalLogger log = InternalLoggerFactory.getInstance(RakNetClient.class);
     private final ClientDatagramHandler handler = new ClientDatagramHandler();
     private final ConcurrentMap<InetSocketAddress, PingEntry> pings = new ConcurrentHashMap<>();
+    private final Set<Consumer<Throwable>> exceptionHandlers = Collections.newSetFromMap(new ConcurrentHashMap<>());
```
Comment (SupremeMortal) [inline raknet/src/main/java/com/nukkitx/network/raknet/RakNetClient.java:27]: Could you use `ConcurrentHashMap.newKeySet()`

File: raknet/src/main/java/com/nukkitx/network/raknet/RakNetServer.java [hunk 4]
```diff
@@ -38,6 +36,7 @@
     private volatile RakNetServerListener listener = null;
     private final int bindThreads;
     private int maxConnections = 1024;
+    private Set<Consumer<Throwable>> exceptionHandler = Collections.newSetFromMap(new ConcurrentHashMap<>());
```
Comment (SupremeMortal) [inline raknet/src/main/java/com/nukkitx/network/raknet/RakNetServer.java:39]: Could you use `ConcurrentHashMap.newKeySet()`

**Review by SupremeMortal — CHANGES_REQUESTED** [review]

**Comment by TobiasG-DE** [comment]
With the latest commit each handler has an individual Id, which can be used to remove specific handlers. That can be used for example if you want to have different exception handlers in different game stages or else

**Review by SupremeMortal — APPROVED** [review]

**Pull request merged**